assets/src/edit-story/utils/useRefreshPostEditURL.js

```diff
@@ -18,26 +18,30 @@
  * External dependencies
  */
 import { useCallback } from 'react';
-import { addQueryArgs } from '@web-stories-wp/design-system';
 
 /**
  * Update page URL in browser.
  *
  * @param {number} postId Current story id.
+ * @param {string} postEditURL Current story's edit link.
  * @return {Function} Function to refresh the post edit URL.
  */
-function useRefreshPostEditURL(postId) {
+function useRefreshPostEditURL(postId, postEditURL) {
   const refreshPostEditURL = useCallback(() => {
-    const getPostEditURL = addQueryArgs('post.php', {
-      post: postId,
-      action: 'edit',
-    });
-    window.history.replaceState(
-      { id: postId },
-      'Post ' + postId,
-      getPostEditURL + window.location.hash
-    );
-  }, [postId]);
+    try {
+      const newUrl = new URL(postEditURL);
+      newUrl.hash = window.location.hash;
+      // Remove the schema and host and just return path, params and hash.
+      const postEditPath = newUrl.toString().replace(newUrl.origin, '');
```

Comment:
Now that there's a `try/catch` this part here should no longer be needed

Can just do 

```js
window.history.replaceState(
  { id: postId },
  'Post ' + postId,
  postEditPath
);
```


By the way, while we're at it, let's remove `'Post ' + postId` and replace it with `''`.

Browsers don't use this field at all it's not even translated anyway.

So final code:

```js
try {
	  const newUrl = new URL(postEditURL);
	  newUrl.hash = window.location.hash;
	  window.history.replaceState(
	    { id: postId },
	    '',
	    newUrl.toString()
	  );
} catch (error) {
  // Do nothing for now.
}
```

untested, but should work

Reply:
No, it doesn't work. This was it change looked like originally. 

https://github.com/google/web-stories-wp/blob/87aa67c851288b362ac00b81ab844db94deca712/assets/src/edit-story/utils/useRefreshPostEditURL.js#L22-L38

This domains the host / schema. This resulted in unit test [fail](https://github.com/google/web-stories-wp/runs/3045283496#step:7:187). 

> SecurityError: replaceState cannot update history to a URL which differs in components other than in path, query, or fragment.

> By the way, while we're at it, let's remove 'Post ' + postId and replace it with ''.
Why are we removing this?

Reply:
> This domains the host / schema. This resulted in unit test fail.

But with the `try/catch` this should no longer fail. Please try it.

> Why are we removing this?

[MDN](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState) says:

> Most browsers currently ignore this parameter, although they may use it in the future. Passing the empty string here should be safe against future changes to the method. Alternatively, you could pass a short title for the state.

So either we remove it or make it properly translatable using `__()`

Reply:
So I made the change and the test still failed. https://github.com/google/web-stories-wp/pull/8268/commits/98074d1eab330b2a47a6299fd0c4c9e0f5b620cc

The only way I got the tests to pass, was to change all the example.com references to localhost. 

I think this might be an issue, for those in multisite that are using domain mapping. What if the url in the REST API is www.example.com and you are editing the site on cms.example.com?